frontend/js/helpers/main.js

```diff
@@ -33,18 +33,20 @@
 customElements.define('gmt-menu', GMTMenu);
 
 const replaceRepoIcon = (uri) => {
-    if (uri.startsWith("https://www.github.com") || uri.startsWith("https://github.com")) {
-        uri = uri.replace("https://www.github.com", '<i class="icon github"></i>');
-        uri = uri.replace("https://github.com", '<i class="icon github"></i>');
-    } else if (uri.startsWith("https://www.bitbucket.com") || uri.startsWith("https://bitbucket.com")) {
-        uri = uri.replace("https://www.bitbucket.com", '<i class="icon bitbucket"></i>');
-        uri = uri.replace("https://bitbucket.com", '<i class="icon bitbucket"></i>');
-    } else if (uri.startsWith("https://www.gitlab.com") || uri.startsWith("https://gitlab.com")) {
-        uri = uri.replace("https://www.gitlab.com", '<i class="icon gitlab"></i>');
-        uri = uri.replace("https://gitlab.com", '<i class="icon gitlab"></i>');
+    const replacements = [
+      { pattern: /^https:\/\/(www\.)?github\.com/, replacement: '<i class="icon github"></i>' },
```

Comment:
The question is though, if that is really wanted.

It will in no case open a security hole, but is still an algorithmic error.

if somebody where to register the domain https://www.github.com.fakedomain.com then it would still replace.

Shall we abide to the CodeQL, so it does not come up again?